spec/mem/tree_borrows/state_machine.md

```diff
@@ -107,10 +133,25 @@ impl LocationState {
         &mut self,
         access_kind: AccessKind,
         node_relation: NodeRelation,
+        protected: Protected,
     ) -> Result {
-        self.permission = self.permission.transition(access_kind, node_relation)?;
+        self.permission = self.permission.transition(access_kind, node_relation, protected, self.accessed)?;
         self.accessed = self.accessed.transition(node_relation);
         ret(())
     }
+
+    /// Create a location state list:
+    /// all locations share the given permission.
+    fn list(permission: Permission, len: Size) -> List<LocationState> {
```

Comment:
Maybe call this `new_list` to make it more clear that this is a constructor?

Also please move it to the top of the impl block.